### DISCUSSION TITLE
Add element in store array

### DISCUSSION THREAD

**Discussion by untlsn** [solidjs/solid#866] [Q&A] [answered]
What's the best way to add a object to an array in store?
```js
setStore('array', [...store.array, newEl])
setStore('array', store.array.length, newEl)
setStore(produce(store => store.array.push(newEl)))
```

**Comment by ryansolid** [comment, marked as answer]
At the end of the list technically 2, 3, 1 in that order for performance. But I wouldn't sweat it. In most other situations 1 over 3. Generally speaking `produce` has a tax per index that changes so cloning the array is just easy/better.

I'd say number 1 is idiomatic but hey if there are places to optimize like 2 go for it. 3 makes certain logic easier to write.

**Reply by jmpedrosa** [reply]
how about a shift?

**Comment by ShivamJoker** [comment]
Guys check out [`toSpliced`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced).
It will do the job for removing the array and also inserting items between array.

```ts
  setCourseStore("sections", selfSecIdx, "chapters", (chap) =>
    chap?.toSpliced(
      selfVidIdx + (isEdgeAtBottom ? 1 : 0),
      0,
      incomingChp[srcVidIdx],
    ),
  );
  setCourseStore("sections", srcSecIdx, "chapters", (chap) =>
    chap?.toSpliced(srcVidIdx, 1),
  );

```
